components/renku_data_services/notebooks/crs.py

```diff
@@ -304,3 +305,27 @@ class AmaltheaSessionV1Alpha1Patch(BaseCRD):
     def to_rfc7386(self) -> dict[str, Any]:
         """Generate the patch to be applied to the session."""
         return self.model_dump(exclude_none=True)
+
+
+def safe_parse_duration(val: Any) -> timedelta:
```

Comment:
Note: I wonder if we should instead mirror the parseDuration code from golang. This is most likely the safest option when reading durations which do not conform to gep-2257.